tiered-storage/file-system/src/main/java/org/apache/bookkeeper/mledger/offload/filesystem/impl/FileStoreBackedReadHandleImpl.java

```diff
@@ -105,30 +105,22 @@ public LedgerMetadata getLedgerMetadata() {
                 promise.completeExceptionally(new BKException.BKIncorrectParameterException());
                 return;
             }
-            long entriesToRead = (lastEntry - firstEntry) + 1;
             List<LedgerEntry> entries = new ArrayList<LedgerEntry>();
-            long nextExpectedId = firstEntry;
             LongWritable key = new LongWritable();
             BytesWritable value = new BytesWritable();
             try {
-                key.set(nextExpectedId - 1);
+                key.set(firstEntry - 1);
                 reader.seek(key);
-                while (entriesToRead > 0) {
-                    reader.next(key, value);
+                reader.next(key, value);
+                long entryId;
+                do {
                     int length = value.getLength();
-                    long entryId = key.get();
-                    if (entryId == nextExpectedId) {
-                        ByteBuf buf = PooledByteBufAllocator.DEFAULT.buffer(length, length);
-                        entries.add(LedgerEntryImpl.create(ledgerId, entryId, length, buf));
-                        buf.writeBytes(value.copyBytes());
-                        entriesToRead--;
-                        nextExpectedId++;
-                    } else if (entryId > lastEntry) {
-                        log.info("Expected to read {}, but read {}, which is greater than last entry {}",
-                                nextExpectedId, entryId, lastEntry);
-                        throw new BKException.BKUnexpectedConditionException();
-                    }
-            }
+                    entryId = key.get();
+                    ByteBuf buf = PooledByteBufAllocator.DEFAULT.buffer(length, length);
```

Comment:
we can't return length is 0 entry, same as above BlockAwareSegmentInputStream.java